### DISCUSSION TITLE
typescript config?

### DISCUSSION THREAD

**Discussion by tve** [Moddable-OpenSource/moddable#1013] [Q&A]
I'm trying to switch to typescript for my moddable projects and am relatively new to typescript. Can anyoneprovide some pointers or samples to get started with the configuration? I'm using vscode. Specifically:

- what is a recommended `tsconfig.json` to put into a project so vscode finds all the right stuff?
- why do the XS tools generate a separate tsconfig? I would prefer to have some guidelines for setting the tsconfig I need up correctly so the tools are happy with it. (I hate magic stuff...)
- is anyone up for submitting a base config for moddable to https://github.com/tsconfig/bases/tree/main/bases ?

I'm currently a bit stuck on how to make an import like `import Timer from "timer"` work in vscode

**Comment by phoddie** [comment]
With some patient help from folks more familiar with TypeScript than me, I have this `tsconfig.json` at the root of my local copy of $MODDABLE. It lets the VS Code editor work with TypeScript source files.

```json
{
    "compilerOptions": {
        "baseUrl": ".",
        "module": "ESNext",
        "target": "ESNext",
        "strict": true,
        "noImplicitAny": true,
        "moduleResolution": "Node",
        "lib": ["ESNext"],
        "paths": {
        },
        "typeRoots": [
            "./typings"
        ]
    }
}
```

**Reply by tve** [reply]
Thanks!!!
The auto-generated tsconfig has:
```
    "lib": ["es2020"],
    "module": "es2020",
    "target": "ES2020",
```
so shouldn't one use the same in projects?
